Source/JavaScriptCore/API/JSValue.h

```diff
@@ -263,39 +300,49 @@ NS_CLASS_AVAILABLE(10_9, 7_0)
 /*!
 @method
 @abstract Convert a JSValue to a double.
-@discussion The JSValue is converted to a number according to the rules specified 
- by the JavaScript language.
 @result The double result of the conversion.
+@discussion Convert the JSValue to a number according to the rules specified by the JavaScript language. Unless the JSValue is a BigInt then this is equivalent to <code>Number(value)</code> in JavaScript.
 */
 - (double)toDouble;
 
 /*!
 @method
 @abstract Convert a JSValue to an <code>int32_t</code>.
-@discussion The JSValue is converted to an integer according to the rules specified 
- by the JavaScript language.
+@discussion The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the JSValue is a BigInt then the value is truncated to an <code>int32_t</code>.
 @result The <code>int32_t</code> result of the conversion.
 */
 - (int32_t)toInt32;
 
 /*!
 @method
 @abstract Convert a JSValue to a <code>uint32_t</code>.
-@discussion The JSValue is converted to an integer according to the rules specified 
- by the JavaScript language.
+@discussion The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the JSValue is a BigInt then the value is truncated to a <code>uint32_t</code>.
 @result The <code>uint32_t</code> result of the conversion.
 */
 - (uint32_t)toUInt32;
 
+/*!
+@method
+@abstract Convert a JSValue to a <code>int64_t</code>.
+@discussion The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the value is a BigInt then the value is truncated to an <code>int64_t</code>.
+*/
+- (int64_t)toInt64 JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@method
+@abstract Convert a JSValue to a <code>uint64_t</code>.
+@discussion The JSValue is converted to an integer according to the rules specified by the JavaScript language. If the value is a BigInt then the value is truncated to a <code>uint64_t</code>.
+*/
+- (uint64_t)toUInt64 JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
 /*!
 @method
 @abstract Convert a JSValue to a NSNumber.
-@discussion If the JSValue represents a boolean, a NSNumber value of YES or NO 
- will be returned. For all other types the value will be converted to a number according 
- to the rules specified by the JavaScript language.
+@discussion If the JSValue represents a boolean, a NSNumber value of YES or NO
+ will be returned. For all other types, the result is equivalent to <code>Number(value)</code> in JavaScript.
```

Comment:
The JS spec says that BigInts cannot be coerced into a Number.  How is that manifested here in the API?

Reply:
> The JS spec says that BigInts cannot be coerced into a Number. 
Yes, this is true. It's still possible to convert BigInt to double. See https://tc39.es/ecma262/#sec-number-constructor-number-value.